### DISCUSSION TITLE
[Suggestion] System, Core/Emulator version indicator on Earned Cheevo Page

### DISCUSSION THREAD

**Discussion by ghost** [RetroAchievements/RAWeb#759] [Ideas]
RetroAchievements is now accessible on many Systems due to the support of RetroArch so it would be nice to have a indicator on the Earned Cheevo Page what System and Core/Emulator version the user has used to unlock the Cheevo.
This is mostly just a decoration thing but could also be used to identify broken Cheevo Code and Incompatible System/Emulators also to create statistics from it (how many earned Cheevos on PS3, Vita, Windows etc.)

- System
A simply System Icon could be displayed next to the Cheevo on earned information Page.
_Example:_ **[PS3]**
_Example:_ **[Vita]**
_Example:_ **[Windows]**

- Emulator/Core
Detailed information of what Core/Emulator version the user has used to unlock the Cheevo.
_Example:_ **RetroArch 1.7.6 - Mesen 0.9.7**
_Example:_ **RASnex9x 0.25**

**Comment by Jamiras** [comment]
See also #814 - similar idea, but tracking compatible hashes.

The standalone emulators already send their version information as part of the User-Agent header. I'm not sure if RetroArch is currently doing anything identifiable. Even if it were, I doubt it would include the core.

**Reply by Jamiras** [reply]
RetroArch has included version information in its User-Agent since 1.8.2 (Dec 2019)

**Comment by meleu** [comment]
Coincidentally I was talking about it with leiradel a few weeks ago. RetroArch sends just `libretro` as User-Agent, [as we can see here](https://github.com/libretro/RetroArch/blob/0f4744bed7cdb2f86e0a533f9ccdc263fa3a270b/libretro-common/net/net_http.c#L458).

We could add more info there, but it needs to be processed on the back-end.

As it requires changes in the database, I think it's going to be a v2 thing.
